app/routes/task_routes.py

```diff
@@ -1 +1,133 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, abort, make_response, request, Response
+from app.models.task import Task
+from ..db import db
+from sqlalchemy import asc, desc
+from datetime import datetime
+import requests
+import json
+import os
+
+bp = Blueprint("tasks_bp", __name__, url_prefix="/tasks")
+
+
+@bp.post("")
+def create_task():
+    request_body = request.get_json()
+    try:
+        new_task = Task.from_dict(request_body)
+    except KeyError as e:
+        response = {"details": "Invalid data"}
+        abort(make_response(response, 400))
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    response = {"task": new_task.to_dict()}
+    return response, 201
+
+
+@bp.get("")
+def get_all_tasks():
+    query = db.select(Task)
+    sort_param = request.args.get("sort")
+    if sort_param == "asc":
+        query = query.order_by(asc(Task.title))
+    if sort_param == "desc":
+        query = query.order_by(desc(Task.title))
+    else:
+        query = query.order_by(Task.id)
+
+    tasks = db.session.scalars(query)
```

Comment:
Prefer to de-dupe this logic by creating a generic function like [`get_models_with_filters`](https://github.com/Ada-C22/flasky/blob/main/app/routes/route_utilities.py#L31) so that you can use it here in `get_all_tasks` and also in the route `get_all_goals`.

With such a refactor, this route would just look like:
```py
@bp.get("")
def get_all_tasks():
        return get_models_with_filters(Task, request.args)